app/controllers/users_controller.rb

```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+
+    if @user.save
+      session[:user_id] = @user.id
+      redirect_to "/"
+    else 
+      flash[:notice] = "Please fill in the required fields"
+      render '/signup'
```

Comment:
We shouldn't hardcode paths, but instead use the Rails built in path helper like 
```ruby 
redirect_to new_user_path
...
else
render :new
```